irodsadmin/management/commands/irodsorphans.py

```diff
@@ -25,184 +26,226 @@
 table_builder = SampleSheetTableBuilder()
 
 
-def get_assay_collections(assays, irods_backend):
-    """Return a list of all assay collection names."""
-    return [irods_backend.get_path(a) for a in assays]
-
-
-def get_assay_subcollections(studies, irods_backend):
-    """Return a list of all assay row colletion names."""
-    collections = []
-    for study in studies:
-        try:
-            study_tables = table_builder.get_study_tables(study)
-        except Exception as ex:
-            logger.error(
-                'Study table building exception for "{}" '
-                'in project "{}" ({}): {}'.format(
-                    study.get_display_name(),
-                    study.investigation.project.title,
-                    study.investigation.project.sodar_uuid,
-                    ex,
-                )
+# Local constants
+DELETED = '<DELETED>'
+ERROR = '<ERROR>'
+
+
+class Command(BaseCommand):
+    """Command to find orphans in iRODS collections."""
+
+    help = 'Find orphans in iRODS project collections.'
+
+    def __init__(self):
+        super().__init__()
+        self.irods_backend = get_backend_api('omics_irods')
+
+    def _sort_colls_on_projects(self, collections):
+        """Helper function to sort collections based on project list"""
+        colls_with_project = []
+        colls_no_project = []
+
+        # Create a set of valid project paths based on project UUIDs
+        valid_project_paths = [
+            self.irods_backend.get_path(p)
+            for p in Project.objects.filter(type=PROJECT_TYPE_PROJECT).order_by(
+                'full_title'
+            )
+        ]
+
+        for coll in collections:
+            uuid_prefix = (
+                coll.path.split('/')[4] if len(coll.path.split('/')) > 4 else ''
```

Comment:
Here, note that we don't necessarily now that the project collection is this deep in the iRODS path. Right now in production and in our local dev setup, the projects collection is in `/sodarZone/projects`, but this may not always be the case (and, indeed, we are likely to change that in production at some point). 

You can get the actual position of the UUID from e.g. calling `IrodsAPI.get_projects_path()` and calculating the depth from there.